python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -119,8 +117,6 @@ def _post_stop_cleanup(future, pg, timeout: Optional[float] = None):
                 f"An exception occurred when trying to stop the Ray actor:"
                 f"{traceback.format_exc()}"
             )
-    finally:
-        remove_placement_group(pg)
```

Comment:
With the new abstraction, the whole lifecycle management of resources should be contained in the resource manager.

This will not leak as we free the resources before we call `_post_stop_cleanup`. Actually, let me double check that `RayTrialExecutor._do_force_trial_cleanup` does not leak anything here, but it should be a simple call to `free_resources`

Reply:
Turns out the force cleanup logic was missing the PG removal, so there was a leak. Thanks for spotting this, I've updated the PR. We free the resources in `_do_force_trial_cleanup` - this still means lifecycle management is contained in the resource manager.

Reply:
We have various places calling `post_stop_cleanup`. What is the reason of calling ResourceManager.free_resources() higher in the calling stack instead of here?